src/pudl/helpers.py

```diff
@@ -1815,10 +1815,10 @@ def assert_cols_areclose(
     # instead of just whether or not there are matches.
     mismatch = df.loc[
         ~np.isclose(
-            df[a_cols],
-            df[b_cols],
+            np.ma.masked_where(np.isnan(df[a_cols]), df[a_cols]),
```

Comment:
Mask all `NaN`'s so if one year's starting balance is `NaN` and previous year's ending balance is not, it will not be treated as a failure